src/sentry/static/sentry/app/components/sidebar/index.jsx

```diff
@@ -285,21 +296,23 @@ class Sidebar extends React.Component {
                     />
                   </Feature>
 
-                  <Feature features={['events-v2']} organization={organization}>
-                    <SidebarItem
-                      {...sidebarItemProps}
-                      onClick={(_id, evt) =>
-                        this.navigateWithGlobalSelection(
-                          generateDiscoverLandingPageRoute(organization.slug),
-                          evt
-                        )
-                      }
-                      icon={<InlineSvg src="icon-telescope" />}
-                      label={t('Discover v2')}
-                      to={generateDiscoverLandingPageRoute(organization.slug)}
-                      id="discover-v2"
-                    />
-                  </Feature>
+                  {discoverVersion === '2' && (
+                    <Feature features={['events-v2']} organization={organization}>
+                      <SidebarItem
```

Comment:
@markstory I think we can set `active` prop to `true` on desired matching routes from React Router for the Query builder and the event details page.

Reply:
Found a different solution. Icon highlighting should be good now.